lib/srv/desktop/rdp/rdpclient/src/lib.rs

```diff
@@ -542,6 +566,12 @@ pub enum CGOErrCode {
     ErrCodeClientPtr = 2,
 }
 
+#[repr(C)]
+pub struct CGOResult {
+    pub err_code: CGOErrCode,
+    pub message: *mut c_char,
+}
```

Comment:
I'm in favor of using this return type everywhere Go calls into Rust, and thereby handling all logging and/or continued error propagation in Go. That way we don't typically need to deal with the issue of rust errors not being properly logged in customer's production environments.

IIRC initially Andrew had something like this, and later I spiked it because it required always remembering to call `free_string`, and I didn't realize the pitfalls of trying to log things on the Rust side. However I think we're good enough with the FFI to keep such things in mind from here on out.

cc @zmb3 @probakowski

Reply:
I agree, it makes sense to always pass error to go side. I'd merge this PR to avoid conflicts and then open another one with next portion of errors ported, WDYT?